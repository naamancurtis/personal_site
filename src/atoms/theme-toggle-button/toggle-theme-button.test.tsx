import React from 'react';
import { shallow } from 'enzyme';
import ThemeToggleButton from './theme-toggler-button.component';

jest.mock('react', () => ({
  // @ts-ignore
  ...jest.requireActual('react'),
  useContext: () => 'test',
}));

describe('<ThemeToggleButton />', () => {
  it('expect to render component', () => {
    const wrapper = shallow(<ThemeToggleButton toggleTheme={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
