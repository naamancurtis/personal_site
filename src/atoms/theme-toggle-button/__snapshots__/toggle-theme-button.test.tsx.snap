// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeToggleButton /> expect to render component 1`] = `
<div>
  <styled.div>
    <styled.input
      checked={false}
      id="toggle-theme"
      onChange={[Function]}
      type="checkbox"
    />
    <styled.label
      htmlFor="toggle-theme"
    >
      <styled.span
        className=""
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "far",
              "lightbulb",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </styled.span>
    </styled.label>
  </styled.div>
</div>
`;
